### PR TITLE
Add redis to docker

### PR DESCRIPTION
**Pull request by rkondratowicz**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ services:
     container_name: send-letter-service
     environment:
       - REDIS_ENABLED
-      - REDIS_HOST
+      - REDIS_HOST=send-letter-redis
       - REDIS_PORT
       - REDIS_PASSWORD
```
Comment (doncem) [inline docker-compose.yml]: should we provide port and password too? or more explicitly: should we provide port and get rid of password?

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ management:
 
 redis:
   enabled: ${REDIS_ENABLED:true}
-  host: ${REDIS_HOST:redis://localhost}
+  host: ${REDIS_HOST:localhost}
```
Comment (doncem) [inline src/main/resources/application.yaml]: if i understood correctly, i think we can provide `redis://send-letter-redis` here as a default, similarly as in postgres configuration

File: docker-compose.yml (hunk 1 above)
Comment (rkondratowicz) [reply docker-compose.yml]: default values from `application.yaml` are used:
```
  port: ${REDIS_PORT:6379}
  password: ${REDIS_PASSWORD:}
```

File: docker-compose.yml [hunk 3]
```diff
@@ -24,3 +24,10 @@ services:
       - REFORM_ENVIRONMENT
     ports:
       - $SERVER_PORT:$SERVER_PORT
+  send-letter-redis:
```
Comment (doncem) [inline docker-compose.yml]: will `send-letter-service` depend on `send-letter-redis`?

File: src/main/resources/application.yaml (hunk 2 above)
Comment (rkondratowicz) [reply src/main/resources/application.yaml]: note that this property is called `host` not `url` or `address`

postgres config has this:
`url: jdbc:postgresql://${DRAFT_STORE_DB_HOST:localhost}:${DRAFT_STORE_DB_PORT:5432}/draftstore`

Comment (doncem) [reply src/main/resources/application.yaml]: ah. so `send-letter-redis` just?

Comment (rkondratowicz) [reply src/main/resources/application.yaml]: yup, full url with port is being built in [`RedisConfiguration.java`](https://github.com/hmcts/send-letter-service/pull/14/files#diff-d53b6cbae124f862a72d9f70a7c6a6e7)

File: docker-compose.yml (hunk 3 above)
Comment (rkondratowicz) [reply docker-compose.yml]: oops, fixed

File: src/main/java/uk/gov/hmcts/reform/sendletter/config/AuthConfiguration.java [hunk 4]
```diff
@@ -21,7 +21,7 @@ public AuthTokenValidator tokenValidator(ServiceAuthorisationApi s2sApi) {
 
     @Bean
     @ConditionalOnProperty(name = "idam.s2s-auth.enabled", havingValue = "false")
-    public AuthTokenValidator tokenValidator() {
+    public AuthTokenValidator tokenValidatorStub() {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/config/AuthConfiguration.java]: although signatures were different, method names were the same for both beans, which was confusing Spring

**Pull request merged**